File: src/Billing.Mediation/Engine/Models/rateassign.cs
// Ported VERBATIM from legacy Models_Mediation/rateassign.cs (MediationModel.rateassign).
// Plain POCO — EF navigation properties (rateplan, rateplanassignmenttuple back-ref) REMOVED so it
// compiles on .NET 8 and deserializes 1:1 from config-manager's legacy-shaped JSON. All scalar fields
// kept verbatim (the rater reads Prefix/Category/SubCategory/startdate/enddate + rateamount/Resolution/
// MinDurationSec/SurchargeTime; the rest ride along for fidelity).
#nullable disable

namespace MediationModel
{
    using System;

    public partial class rateassign
    {
        public int id { get; set; }
        public int Prefix { get; set; }
        public string description { get; set; }
        public decimal rateamount { get; set; }
        public int WeekDayStart { get; set; }
        public int WeekDayEnd { get; set; }
        public string starttime { get; set; }
        public string endtime { get; set; }
        public int Resolution { get; set; }
        public float MinDurationSec { get; set; }
        public int SurchargeTime { get; set; }
        public decimal SurchargeAmount { get; set; }
        public Nullable<long> idrateplan { get; set; }
        public string CountryCode { get; set; }
        public Nullable<System.DateTime> date1 { get; set; }
        public Nullable<int> field1 { get; set; }
        public Nullable<int> field2 { get; set; }
        public int field3 { get; set; }
        public string field4 { get; set; }
        public string field5 { get; set; }
        public System.DateTime startdate { get; set; }
        public Nullable<System.DateTime> enddate { get; set; }
        public int Inactive { get; set; }
        public int RouteDisabled { get; set; }
        public int Type { get; set; }
        public int Currency { get; set; }
        public Nullable<float> OtherAmount1 { get; set; }
        public Nullable<float> OtherAmount2 { get; set; }
        public Nullable<float> OtherAmount3 { get; set; }
        public Nullable<decimal> OtherAmount4 { get; set; }
        public Nullable<decimal> OtherAmount5 { get; set; }
        public Nullable<float> OtherAmount6 { get; set; }
        public Nullable<float> OtherAmount7 { get; set; }
        public Nullable<float> OtherAmount8 { get; set; }
        public Nullable<float> OtherAmount9 { get; set; }
        public Nullable<float> OtherAmount10 { get; set; }
        public decimal TimeZoneOffsetSec { get; set; }
        public Nullable<int> RatePosition { get; set; }
        public Nullable<float> IgwPercentageIn { get; set; }
        public string ConflictingRateIds { get; set; }
        public Nullable<long> ChangedByTaskId { get; set; }
        public Nullable<System.DateTime> ChangedOn { get; set; }
        public Nullable<int> Status { get; set; }
        public Nullable<long> idPreviousRate { get; set; }
        public Nullable<sbyte> EndPreviousRate { get; set; }
        public Nullable<sbyte> Category { get; set; }
        public Nullable<sbyte> SubCategory { get; set; }
        public Nullable<int> ChangeCommitted { get; set; }
        public string ConflictingRates { get; set; }
        public string OverlappingRates { get; set; }
        public string Comment1 { get; set; }
        public string Comment2 { get; set; }
        public string BillingParams { get; set; }
    }
}

File: java/src/main/java/com/telcobright/billing/mediation/engine/models/rateassign.java
// Ported VERBATIM from legacy Models_Mediation/rateassign.cs (MediationModel.rateassign).
// Plain POCO — EF navigation properties (rateplan, rateplanassignmenttuple back-ref) REMOVED so it
// compiles on .NET 8 and deserializes 1:1 from config-manager's legacy-shaped JSON. All scalar fields
// kept verbatim (the rater reads Prefix/Category/SubCategory/startdate/enddate + rateamount/Resolution/
// MinDurationSec/SurchargeTime; the rest ride along for fidelity).
package com.telcobright.billing.mediation.engine.models;

import java.math.BigDecimal;
import java.time.LocalDateTime;

public class rateassign {
    public int id;
    public int Prefix;
    public String description;
    public BigDecimal rateamount;
    public int WeekDayStart;
    public int WeekDayEnd;
    public String starttime;
    public String endtime;
    public int Resolution;
    public float MinDurationSec;
    public int SurchargeTime;
    public BigDecimal SurchargeAmount;
    public Long idrateplan;
    public String CountryCode;
    public LocalDateTime date1;
    public Integer field1;
    public Integer field2;
    public int field3;
    public String field4;
    public String field5;
    public LocalDateTime startdate;
    public LocalDateTime enddate;
    public int Inactive;
    public int RouteDisabled;
    public int Type;
    public int Currency;
    public Float OtherAmount1;
    public Float OtherAmount2;
    public Float OtherAmount3;
    public BigDecimal OtherAmount4;
    public BigDecimal OtherAmount5;
    public Float OtherAmount6;
    public Float OtherAmount7;
    public Float OtherAmount8;
    public Float OtherAmount9;
    public Float OtherAmount10;
    public BigDecimal TimeZoneOffsetSec;
    public Integer RatePosition;
    public Float IgwPercentageIn;
    public String ConflictingRateIds;
    public Long ChangedByTaskId;
    public LocalDateTime ChangedOn;
    public Integer Status;
    public Long idPreviousRate;
    public Byte EndPreviousRate;
    public Byte Category;
    public Byte SubCategory;
    public Integer ChangeCommitted;
    public String ConflictingRates;
    public String OverlappingRates;
    public String Comment1;
    public String Comment2;
    public String BillingParams;
    // The legacy EF nav rateplanassignmenttuple's underlying FK column — the tuple this rate belongs
    // to (used by the RateCache loader to group rates per tuple).
    public Integer idrateplanassignmenttuple;
}
